### ISSUE TITLE
release obo file 4.1.31 has windows line endings

### ISSUE THREAD

**Issue by mwalzer** [HUPO-PSI/psi-ms-CV#48]
For https://github.com/HUPO-PSI/psi-ms-CV/releases/download/4.1.31/psi-ms.obo , lines end in '\r\n'.
I think we don't want that.

**Comment by edeutsch** [comment]
But this is fixed in the latest version 4.1.45? It seems to be.
Should we just create a new release to fix this?
It is perhaps not good that our more recent release is so way behind the master branch.
Should we create a policy for creating releases more often?

**Comment by mwalzer** [comment]
@edeutsch you thinking of something like: after at most either 3 new terms, or 3 months past, or 10 changed entries?
No windows line ends in the master branch as far as I could see.

**Comment by chambm** [comment]
Why bother with releases at all? Either release for every update or get rid of releases entirely and just point at master. It's just a single text file.

**Comment by edeutsch** [comment]
I agree that we don't need releases, working from master should be fine. I always do.
BUT, we already have some releases. And the most recent one is slightly broken.
Some people have a policy to use the latest release rather than master/trunk

I think we should either:
- Unrelease the 3 releases we already have, so there are no releases. Can we even do that? And tell everyone to use master.
or
- Update the release to the latest AND then create a calendar reminder to release every 2 months or so. I could do that.

Thoughts?

**Comment by chambm** [comment]
There's a delete button for the tag, I'm not sure what it will do to the release if you delete the tag that backs it.

**Comment by edeutsch** [comment]
Well, what does everyone think? Try to nuke all the releases or try to be disciplined about periodic releases?

**Comment by mwalzer** [comment]
I do agree that releases are somewhat pointless except for getting a sort-of stable URL for the obo file. However I'd be wary of using our main branch as a rolling release branch, unless we have a neat way of vetting new changes. With the GitHub actions we have the basis to automate that already in place. Maybe we could bring @althonos in to see if it is possible to create a sound way to automate the vetting?

**Comment by chambm** [comment]
It's easy but slightly convoluted, but anyone can point at any commit to get a stable URL. E.g.
https://github.com/HUPO-PSI/psi-ms-CV/blob/5e02c611b8392301ce00e8d7bb3a4d63848ef8b3/psi-ms.obo

There's nothing wrong with using master as a rolling release, because we certainly should be vetting PRs before they are merged. Merging a PR should mean that the changes are accepted and ready for release.

**Comment by mwalzer** [comment]
> point at any commit to get a stable URL. E.g.

Yes, while the result is slightly cryptic to most not familiar with git this will surely work. Arguing from a publications' methods section point, so to speak, it would certainly be an easing if we had regular interval versions, just to simplify the process of navigationg through the progress of the obo. I have no idea however what accepted norms there are for this in the field of ontologies in general... anyone?

My point about using a separate release branch is that I suppose that doing releases involved an extra effort of controlling consistency, completeness, and a general eye on the obo as a whole. And while vetting PRs is surely a must, I'm not so sure if all those points mentioned before would be or could be always observed for each PR. Unless we have a proper CI that does all these steps for us automatically.

**Comment by althonos** [comment]
Here is the OBO theory about the data version: If you specifiy a data version in your ontology, it is normally safe to expand it to a PURL to the OBO library. 

For instance, the GO ontology has a rolling release (that you get from http://purl.obolibrary.org/obo/go.obo) but a persistent version can be reached through purl.obolibrary.org/obo/go/releases/2019-07-01/go.obo to get the with data version `releases/2019-07-01`. This is something that can be configured within the PURL system (https://github.com/OBOFoundry/purl.obolibrary.org/blob/master/config/go.yml#L55). GO serves it from their own releases server, but it's trivial to use the GitHub release URLs instead.

I guess you could have a workflow that does periodic releases using the GitHub actions, something along the lines of:
- figure out the new data version (using semver) and bump it automatically
- tag the commit using the new data version
- validate the file 
- make a release and attach the file to the release so that it can be reached with a persistent URL

As @mwalzer suggested, I can probably help with the CI if you need some more vetting.

**Comment by chambm** [comment]
The vast majority of CV updates here are new terms. Shouldn't new terms get a new "release" as long as they approved at the time of the PR? I suppose a few PRs that are obsoletions or definition fixes don't strictly need an immediate "release", but I don't think it hurts either. Unfortunately we don't have a CV maintainer that will give "extra effort of controlling consistency, completeness, and a general eye on the obo as a whole." I suspect we'll just need to fix issues with the CI as soon as we discover it missed a problem with some merged PR.
